provision.sh

```diff
@@ -10,6 +16,19 @@ package() {
     sudo apt-get -y install "$@"
 }
 
+git_clone() {
+    # Clone git repo into directory, clearing directory if it exists
+    # Usage: git_clone URL DIR [GIT_CLONE OPTIONS...]
+
+    local url="$1"
+    shift
+    local dir="$1"
```

Comment:
I added the `git_clone` function so that it could remove the destination directory before `git clone`.  I was running into errors when running git clone after the destination directory already existed. Because you cannot determine the name of the default destination directory from just the git URL, I needed to have the destination directory as a parameter.

This allows you to run the provision script multiple times.